v2.0.25 fails with 404 when using {repository} URL parameter set to ${{ github.repository }}

We have a workflow that requests the list of files modified by a PR. Here's the code:
```yaml
name: kube-score
on:
  # pull_request workflows run in the context of the fork, when applicable, and
  # thus does not have permission to comment on the PR. To support this, this
  # workflow only publishes an artifact and the comment will be posted by the
  # pr-comment workflow.
  # https://github.blog/2020-08-03-github-actions-improvements-for-fork-and-pull-request-workflows/#improvements-for-public-repository-forks
  pull_request:
    paths:
      - "**-version.txt"
jobs:
  kube-score:
    runs-on: ubuntu-latest
    timeout-minutes: 10
    steps:
      - uses: octokit/request-action@v2.x
        id: list_files
        with:
          route: GET /repos/:repository/pulls/:number/files
          repository: ${{ github.repository }}
          number: ${{ github.event.pull_request.number }}
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
This stopped working when v2.0.25 was released. The error is `not found`:
```
2021-01-25T21:49:32.3761923Z Download action repository 'octokit/request-action@v2.x'
2021-01-25T21:49:35.9635012Z ##[warning]Unexpected input(s) 'repository', 'number', valid inputs are ['route', 'mediaType']
2021-01-25T21:49:35.9691635Z ##[group]Run octokit/request-action@v2.x
2021-01-25T21:49:35.9692287Z with:
2021-01-25T21:49:35.9692842Z   route: GET /repos/:repository/pulls/:number/files
2021-01-25T21:49:35.9693621Z   repository: <redacted>
2021-01-25T21:49:35.9694284Z   number: 8631
2021-01-25T21:49:35.9694703Z   mediaType: {}
2021-01-25T21:49:35.9695128Z env:
2021-01-25T21:49:35.9696104Z   GITHUB_TOKEN: ***
2021-01-25T21:49:35.9696545Z ##[endgroup]
2021-01-25T21:49:36.3260397Z GET /repos/:repository/pulls/:number/files
2021-01-25T21:49:36.3261858Z > repository: <redacted>
2021-01-25T21:49:36.3262527Z > number: 8631
2021-01-25T21:49:36.3264255Z > mediaType: [object Object]
2021-01-25T21:49:36.4425840Z ##[error]Not Found
```
I looked at [the diff](https://github.com/octokit/request-action/compare/v2.0.24...v2.0.25) and it's not obvious why it broke. I switched to `- uses: octokit/request-action@v2.0.24` and it started working again.

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -30,9 +30,6 @@ jobs:
           route: POST /repos/{owner}/{repo}/check-runs
           owner: octokit
           repo: request-action
-          mediaType: | # The | is significant!
-            previews: 
-              - antiope
           name: "Test check run"
           head_sha: ${{ github.sha }}
           output: | # The | is significant!
@@ -52,9 +49,6 @@ jobs:
           route: PATCH /repos/{owner}/{repo}/check-runs/{check_run_id}
           owner: octokit
           repo: request-action
-          mediaType: | # The | is significant!
-            previews: 
-              - antiope
           check_run_id: ${{ fromJson(steps.create_check_run.outputs.data).id }}
           conclusion: "success"
           status: "completed"

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -21,6 +21,7 @@ jobs:
           owner: octokit
           repo: request-action
       - run: "echo latest release: '${{ steps.get_latest_release.outputs.data }}'"
+
       # Create check run
       - name: "Create check run for ${{ github.sha }}"
         uses: ./
@@ -57,5 +58,14 @@ jobs:
           check_run_id: ${{ fromJson(steps.create_check_run.outputs.data).id }}
           conclusion: "success"
           status: "completed"
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      # See https://github.com/octokit/request-action/issues/71
+      - name: "Un-encode {repo} URL parameter when it's set to github.repository"
+        uses: ./
+        with:
+          route: GET /repos/{repository}
+          repository: ${{ github.repository }}
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

--- a/README.md
+++ b/README.md
@@ -50,9 +50,6 @@ jobs:
           route: POST /repos/{owner}/{repo}/check-runs
           owner: octokit
           repo: request-action
-          mediaType: | # The | is significant!
-            previews: 
-              - antiope
           name: "Test check run"
           head_sha: ${{ github.sha }}
           output: | # The | is significant!
@@ -71,9 +68,6 @@ jobs:
           route: PATCH /repos/{owner}/{repo}/check-runs/{check_run_id}
           owner: octokit
           repo: request-action
-          mediaType: | # The | is significant!
-            previews: 
-              - antiope
           check_run_id: ${{ fromJson(steps.create_check_run.outputs.data).id }}
           conclusion: "success"
           status: "completed"

--- a/index.js
+++ b/index.js
@@ -18,10 +18,14 @@ async function main() {
 
     // workaround for https://github.com/octokit/request-action/issues/71
     // un-encode "repo" in /repos/{repo} URL when "repo" parameter is set to ${{ github.repository }}
-    const options = octokit.request.endpoint(route, parameters);
+    const { url, body, ...options } = octokit.request.endpoint(
+      route,
+      parameters
+    );
     const requestOptions = {
       ...options,
-      url: options.url.replace(
+      data: body,
+      url: url.replace(
         /\/repos\/([^/]+)/,
         (_, match) => "/repos/" + decodeURIComponent(match)
       ),

--- a/index.js
+++ b/index.js
@@ -16,26 +16,24 @@ async function main() {
       core.info(`> ${name}: ${value}`);
     }
 
-    core.debug(`route: ${inspect(route)}`);
-    core.debug(`parameters: ${inspect(parameters)}`);
-    core.debug(
-      `parsed request options: ${inspect(
-        octokit.request.endpoint(route, parameters)
-      )}`
-    );
-
-    const time = Date.now();
-
     // workaround for https://github.com/octokit/request-action/issues/71
     // un-encode "repo" in /repos/{repo} URL when "repo" parameter is set to ${{ github.repository }}
     const options = octokit.request.endpoint(route, parameters);
-    const { status, headers, data } = await octokit.request({
+    const requestOptions = {
       ...options,
       url: options.url.replace(
         /\/repos\/([^/]+)/,
         (_, match) => "/repos/" + decodeURIComponent(match)
       ),
-    });
+    };
+
+    core.debug(`route: ${inspect(route)}`);
+    core.debug(`parameters: ${inspect(parameters)}`);
+    core.debug(`parsed request options: ${inspect(requestOptions)}`);
+
+    const time = Date.now();
+
+    const { status, headers, data } = await octokit.request(requestOptions);
 
     core.info(`< ${status} ${Date.now() - time}ms`);
 

--- a/index.js
+++ b/index.js
@@ -25,7 +25,17 @@ async function main() {
     );
 
     const time = Date.now();
-    const { status, headers, data } = await octokit.request(route, parameters);
+
+    // workaround for https://github.com/octokit/request-action/issues/71
+    // un-encode "repo" in /repos/{repo} URL when "repo" parameter is set to ${{ github.repository }}
+    const options = octokit.request.endpoint(route, parameters);
+    const { status, headers, data } = await octokit.request({
+      ...options,
+      url: options.url.replace(
+        /\/repos\/([^/]+)/,
+        (_, match) => "/repos/" + decodeURIComponent(match)
+      ),
+    });
 
     core.info(`< ${status} ${Date.now() - time}ms`);
 

